### PR TITLE
Add filestore_snapshot resource

### PR DESCRIPTION
**Pull request by roaks3**
Adds the `filestore_snapshot` resource for taking snapshots of enterprise filestore instances

b/227499956

Example:
```
resource "google_filestore_snapshot" "my_snapshot" {
  name     = "my-snapshot"
  instance = google_filestore_instance.my_instance.name
  location = "us-central1"
}
```

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
google_filestore_snapshot
```

### CODE REVIEW

File: mmv1/products/filestore/terraform.yaml [hunk 1]
```diff
@@ -53,6 +53,22 @@ overrides: !ruby/object:Overrides::ResourceOverrides
     custom_code: !ruby/object:Provider::Terraform::CustomCode
       constants: templates/terraform/constants/filestore.erb
       pre_create: templates/terraform/pre_create/filestore_instance.go.erb
+  Snapshot: !ruby/object:Overrides::Terraform::ResourceOverride
+    mutex: "filestore/{{project}}"
+    error_retry_predicates: ["isNotFilestoreQuotaError"]
```
Comment (roaks3) [inline mmv1/products/filestore/terraform.yaml:58]: I'm assuming that snapshots will be subject to the same problem as filestore instances: if the quota is reached, there is no sense in retrying.

File: mmv1/products/filestore/api.yaml [hunk 2]
```diff
@@ -253,3 +253,72 @@ objects:
...
+        required: true
+        input: true
+        url_param_only: true
+        pattern: projects/{{project}}/locations/{{location}}/instances/{{instance}}/snapshots/{{name}}
+      - !ruby/object:Api::Type::String
+        name: 'description'
+        description: |
+          A description of the snapshot with 2048 characters or less. Requests with longer descriptions will be rejected.
+      - !ruby/object:Api::Type::Enum
+        name: 'state'
```
[the remark this answers is not in the document]
Comment (shuyama1) [reply mmv1/products/filestore/api.yaml:308]: Keeping this field seems fine!

**Review by shuyama1 — APPROVED** [review]
LGTM!

**Review by shuyama1 — APPROVED** [review]
Perfect! Thanks!

**Pull request merged**